control-plane/config-entries/controllersv2/mesh_gateway_controller.go

```diff
@@ -124,7 +125,28 @@ func (r *MeshGatewayController) onCreateUpdate(ctx context.Context, req ctrl.Req
 		return fmt.Errorf("unable to create role binding: %w", err)
 	}
 
-	// TODO NET-6393
+	//Create Service
+
+	mergeServiceOp := func(ctx context.Context, existingObject, object client.Object) error {
+		existingService, ok := existingObject.(*corev1.Service)
+		if !ok && existingService != nil {
+			return fmt.Errorf("unable to infer existing service type")
+		}
+		builtService, ok := object.(*corev1.Service)
+		if !ok {
+			return fmt.Errorf("unable to infer built service type")
+		}
+
+		mergedService := mergeService(existingService, builtService)
+
+		_, err := controllerutil.CreateOrUpdate(ctx, r.Client, mergedService, func() error { return nil })
```

Comment:
Out of curiosity, what is the role of this no-op function?

Reply:
I guess it does carry out an action `CreateOrUpdate creates or updates the given object in the Kubernetes
cluster`, here I just don't log the result of the operation.

Reply:
Ah! I should have been clearer. The `func() error { return nil }` no-op function.